### PR TITLE
Vision Tests + Jenkins Improvements

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
1. Run `pytest`s marked with `@pytest.mark.vision` on the mosaicml vision dockerimage.
2. Run gpu and extraDeps=dev tests on the minimum version of python3.7 + torch1.9 to ensure that we do not depend on new functionality.
3. Upgrade the vision dockerimage to python 3.9 for FFCV compatibility (FFCV is incompatible with python 3.7)
4. (Unrelated to this PR but coming along for the ride) Log non-rank-zero stdout and stderr as Jenkins artifacts. Should make debugging failures of multi-rank tests easier.

### CODE REVIEW

**Review by dskhudia — APPROVED** [review]
I didn't look at Jenkins file changes but other changes look good.

File: docker/pytorch/build_matrix.sh [hunk 1]
```diff
@@ -12,4 +12,4 @@ echo "TAG='mosaicml/pytorch:1.10.0_cpu-python3.9-ubuntu20.04' BASE_IMAGE='ubuntu
...
 echo "TAG='mosaicml/pytorch:1.9.1_cpu-python3.8-ubuntu18.04' BASE_IMAGE='ubuntu:18.04' PYTHON_VERSION='3.8' CUDA_VERSION_TAG='cpu' CUDA_VERSION='cpu' LINUX_DISTRO='ubuntu:18.04' PYTORCH_VERSION='1.9.1' TORCHVISION_VERSION='0.10.1' TARGET='pytorch_stage'"
 # MMCV_TORCH_VERSION for the link https://download.openmmlab.com/mmcv/dist/${CUDA_VERSION_TAG}/${MMCV_TORCH_VERSION}/index.html
-echo "TAG='mosaicml/pytorch_vision:1.9.1_cu111-python3.7-ubuntu20.04' TAG='mosaicml/pytorch_vision:latest' BASE_IMAGE='nvidia/cuda:11.1.1-cudnn8-devel-ubuntu20.04' PYTHON_VERSION='3.7' CUDA_VERSION_TAG='cu111' CUDA_VERSION='11.1.1' LINUX_DISTRO='ubuntu:20.04' PYTORCH_VERSION='1.9.1' TORCHVISION_VERSION='0.10.1' TARGET='vision_stage' MMCV_TORCH_VERSION=torch1.9.0"
+echo "TAG='mosaicml/pytorch_vision:1.9.1_cu113-python3.9-ubuntu20.04' TAG='mosaicml/pytorch_vision:latest' BASE_IMAGE='nvidia/cuda:11.3.1-cudnn8-devel-ubuntu20.04' PYTHON_VERSION='3.9' CUDA_VERSION_TAG='cu113' CUDA_VERSION='11.3.1' LINUX_DISTRO='ubuntu:20.04' PYTORCH_VERSION='1.10.0' TORCHVISION_VERSION='0.11.1' TARGET='vision_stage' MMCV_TORCH_VERSION=torch1.10.0"
```
Comment (bandish-shah) [inline docker/pytorch/build_matrix.sh:15]: The vision tests aren't running, this should be torch `1.10.0`

**Review by bandish-shah — CHANGES_REQUESTED** [review]
Please fix the issue with vision tests not being run.

File: docker/pytorch/build_matrix.sh (hunk 1 above)
Comment (ravi-mosaicml) [reply docker/pytorch/build_matrix.sh]: Good catch, thanks!

**Review by bandish-shah — APPROVED** [review]
LGTM

**Pull request merged**